### DISCUSSION TITLE
How to correctly type pre-rendered data using InfiniteQuery in Next.js

### DISCUSSION THREAD

**Discussion by patrick-xin** [TanStack/query#3349] [Q&A] [answered]
Hello everyone! I was playing around with InfiniteQuery in a Next.js app. I want to pre-render some posts in getServerSideProps, then on the client using InfiniteQuery fetching the rest posts. Posts data looks like this:
![Screen Shot 2022-02-28 at 13 32 04](https://user-images.githubusercontent.com/81268941/156062272-dbfa2ecb-1dac-4bfa-a10b-5c7eb67adbb1.png)
![Screen Shot 2022-02-28 at 13 34 44](https://user-images.githubusercontent.com/81268941/156062433-749bfff4-c9a6-4417-806b-4b125661605f.png)
 How do I correctly type posts.posts? Thank you!

**Comment by TkDodo** [comment, marked as answer]
`InfiniteData` contains `pages` and `pageParams`. So it's:

```
posts.pages.flatMap(page => page.posts.map(post => ...))
```

**Reply by patrick-xin** [reply]
Got it! Thank you so much!
